drivers/cleanup.py

```diff
@@ -147,22 +147,35 @@ def runAbortable(func, ret, ns, abortTest, pollInterval, timeOut):
         pid = os.fork()
         if pid:
             startTime = time.time()
-            while True:
-                if resultFlag.test("success"):
-                    Util.log("  Child process completed successfully")
-                    resultFlag.clear("success")
-                    return
-                if resultFlag.test("failure"):
-                    resultFlag.clear("failure")
-                    raise util.SMException("Child process exited with error")
-                if abortTest() or abortSignaled:
-                    os.killpg(pid, signal.SIGKILL)
-                    raise AbortException("Aborting due to signal")
-                if timeOut and time.time() - startTime > timeOut:
-                    os.killpg(pid, signal.SIGKILL)
-                    resultFlag.clearAll()
-                    raise util.SMException("Timed out")
-                time.sleep(pollInterval)
+            try:
+                while True:
+                    if resultFlag.test("success"):
+                        Util.log("  Child process completed successfully")
+                        resultFlag.clear("success")
+                        return
+                    if resultFlag.test("failure"):
+                        resultFlag.clear("failure")
+                        raise util.SMException("Child process exited with error")
+                    if abortTest() or abortSignaled:
+                        os.killpg(pid, signal.SIGKILL)
+                        raise AbortException("Aborting due to signal")
+                    if timeOut and time.time() - startTime > timeOut:
+                        os.killpg(pid, signal.SIGKILL)
+                        resultFlag.clearAll()
+                        raise util.SMException("Timed out")
+                    time.sleep(pollInterval)
+            finally:
+                wait_pid = 0
+                rc = -1
+                count = 0
```

Comment:
Could we simplify the code by having a resultFlag.clearAll() in the finally?

Reply:
Actually, the whole resultflag piece can be removed, but that's a bigger change for another day. We can just use the exit code from the process by moving the `waitpid(pid, WNOHANG)` inside the loop, basically poll the subprocess until it completes and the read the exit status.